spirit/core/utils/markdown/block.py

```diff
@@ -41,12 +41,22 @@ class BlockGrammar(mistune.BlockGrammar):
     # * https://www.youtube.com/watch?v=Z0UISCEe52Y
     # * http://youtu.be/afyK1HSFfgw
     # * https://www.youtube.com/embed/vsF0K3Ou1v0
+    #
+    # Also works for timestamps:
+    # * https://www.youtube.com/watch?v=Z0UISCEe52Y&t=1m30s
+    # * https://www.youtube.com/watch?v=O1QQajfobPw&t=1h1m38s
+    # * https://www.youtube.com/watch?v=O1QQajfobPw&feature=youtu.be&t=3698
+    # * https://youtu.be/O1QQajfobPw?t=3698
+    # * https://youtu.be/O1QQajfobPw?t=1h1m38s
+    #
     youtube = re.compile(
         r'^https?://(www\.)?'
         r'(youtube\.com/watch\?v='
         r'|youtu\.be/'
         r'|youtube\.com/embed/)'
         r'(?P<id>[a-zA-Z0-9_\-]{11})'
+        r'((&|\?)?feature=youtu.be)?'
```

Comment:
> & or \? are optional because one case uses ?v= but the others don't

Yes, but one of them must be present. If you make them optional then you are saying this is valid: `youtube.com/foo/?v=bart=123` (notice the missing &).